### PR TITLE
Move protection update

### PR DESCRIPTION
**Pull request by milyin**
- updated to latest zenoh-c API version: move protection structures `z_xxx_moved_t` and made `z_check`/`z_null` internal
- updates to make zenoh-cpp compilable on windows machine with MSVC compiler

### CODE REVIEW

File: examples/universal/z_get.cxx [hunk 1]
```diff
@@ -86,7 +86,15 @@ int _main(int argc, char **argv) {
         done_signal.notify_all();
     };
 
+#if __cplusplus >= 201703L
```
Comment (yellowhatter) [inline examples/universal/z_get.cxx:89]: I thought we declare C++ 17 and higher, why do we need this check?

**Review by yellowhatter — APPROVED** [review]

File: examples/universal/z_get.cxx (hunk 1 above)
Comment (yellowhatter) [reply examples/universal/z_get.cxx:89]: (here and many places below)

Comment (milyin) [reply examples/universal/z_get.cxx:89]: Under MSVC on windows the compiler refuses this aggregate initialization syntax 
```c++
session.get(keyexpr, "", on_reply, on_done, {.target = Z_QUERY_TARGET_ALL, .payload = Bytes::serialize(value)});
```
which is C++20 only
https://en.cppreference.com/w/cpp/language/aggregate_initialization

But it works in clang and gcc even in c++17 mode. I think it's good to keep this c++20 syntax in examples to demonstrate the best way to do the things, but we also need to provide fully compatible solution.

Comment (yellowhatter) [reply examples/universal/z_get.cxx:89]: Ok, got it!

File: include/zenoh/detail/interop.hxx [hunk 2]
```diff
@@ -68,6 +68,11 @@ auto loan(OwnedType& o) {
     return ::z_loan_mut(*as_owned_c_ptr(o));
 }
 
+template <class OwnedType>
+auto move_(OwnedType& o) {
```
Comment (DenisBiryukov91) [inline include/zenoh/detail/interop.hxx:72]: I would suggest to rename this function into as_moved_c_obj  or to_moved_c_obj

**Review by DenisBiryukov91 — APPROVED** [review]

**Pull request merged**